crates/rspack_core/src/compiler/module_executor/mod.rs

```diff
@@ -210,12 +236,26 @@ impl ModuleExecutor {
           original_module_context.unwrap_or(Context::from("")),
         );
         let dep_id = *dep.id();
-        v.insert(dep_id);
+        v.insert(DepStatus {
+          id: dep_id,
+          should_update: false,
+        });
         (ExecuteParam::Entry(Box::new(dep), layer.clone()), dep_id)
       }
-      Entry::Occupied(v) => {
-        let dep_id = *v.get();
-        (ExecuteParam::DependencyId(dep_id), dep_id)
+      Entry::Occupied(mut v) => {
+        let dep_status = v.get_mut();
+        let dep_id = dep_status.id;
+        if dep_status.should_update {
+          let dep = LoaderImportDependency::new_with_id(
+            dep_id,
+            request.clone(),
+            original_module_context.unwrap_or(Context::from("")),
+          );
+          dep_status.should_update = false;
+          (ExecuteParam::Entry(Box::new(dep), layer.clone()), dep_id)
```

Comment:
Since dep.id is already registered, we can replace it with `Execute Param::Dependency` here.
https://github.com/web-infra-dev/rspack/blob/37e17cf03489ef20b5907701f562399aac29e3e4/crates/rspack_core/src/compiler/module_executor/ctrl.rs#L162-L174

Reply:
Sorry I'm not sure what does it mean...

Reply:
My mistake, ignore it